inspectit-oce-agent/build.gradle

```diff
@@ -21,4 +21,116 @@ jar {
     from {
         configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
     }
-}
\ No newline at end of file
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+// Based on the integration tests of https://github.com/census-instrumentation/opencensus-java/tree/master/contrib/agent
+//----------------------------------------------------------------------------------------------------------------------
+
+sourceSets {
+    systemTest {
+        java {
+            srcDir file('src/system-test/java')
+        }
+        resources.srcDir file('src/system-test/resources')
+    }
+}
+
+configurations {
+    systemTestCompile.extendsFrom testImplementation
+}
+
+dependencies {
+
+    systemTestImplementation(
+            "io.opencensus:opencensus-testing:${openCensusVersion}",
+            "io.opencensus:opencensus-api:${openCensusVersion}",
+            'org.junit.jupiter:junit-jupiter-engine:5.3.1',
+            'org.junit.jupiter:junit-jupiter-api:5.3.1',
+            'org.assertj:assertj-core:3.11.1',
+
+            'org.apache.httpcomponents:httpclient:4.5.6'
+    )
+}
+
+if (!project.hasProperty("systemTestJavaHomes")) {
+    ext.systemTestJavaHomes = "$System.env.JAVA_HOME"
+}
+
+task systemTest {
+
+    def javaExecutables = ("${systemTestJavaHomes}")
+            .tokenize(File.pathSeparator)
+            .collect { org.apache.tools.ant.taskdefs.condition.Os.isFamily(
+            org.apache.tools.ant.taskdefs.condition.Os.FAMILY_WINDOWS)
+            ? "${it}/bin/java.exe"
+            : "${it}/bin/java" }
+    .collect { new File(it).getCanonicalPath() }
+            .unique()
+
+    assert javaExecutables.size > 0 :
+            'No Java executables found for running system tests'
+
+    def agentJarPath = jar.archivePath.toString()
+
+    javaExecutables.eachWithIndex { javaExecutable, index ->
+        def perVersionSystemTest = task("systemTest_${index}", type: Test) {
```

Comment:
nice stuff, can we maybe update the name of this new task to something like:

```
        def paths = "${javaExecutable}".split(File.separator)
        def javaName = paths.length > 2 ? paths[paths.length - 3] : "${index}"
        def perVersionSystemTest = task("systemTest_${javaName}", type: Test) {
```
this way I would have task as `systemTest_java-8-oracle` instead of  `systemTest_0`

Reply:
does not work when you have two jre paths: `/usr/lib/jvm/java-8-openjdk-amd64/jre:/usr/lib/jvm/java-8-oracle/jre`, any idea how to solve this? if not let's keep it how it was :cry:

Reply:
But I run this and it was all green:

`../gradlew  -PsystemTestJavaHomes="/usr/lib/jvm/java-11-openjdk-amd64:/usr/lib/jvm/java-8-openjdk-amd64:/usr/lib/jvm/java-8-oracle:/usr/lib/jvm/java-9-oracle" check`